plutus-executables/src/AgdaUnparse.hs

```diff
@@ -0,0 +1,135 @@
+module AgdaUnparse where
+
+import Data.ByteString (ByteString)
+import Data.Functor.Identity
+import Data.Text (Text)
+import Data.Text qualified as T
+import PlutusCore qualified as PLC
+import PlutusCore.Crypto.BLS12_381.G1 qualified as BLS12_381.G1
+import PlutusCore.Crypto.BLS12_381.G2 qualified as BLS12_381.G2
+import PlutusCore.Crypto.BLS12_381.Pairing qualified as BLS12_381.Pairing
+import PlutusCore.Data (Data)
+import PlutusCore.Data qualified as Data
+import PlutusCore.Default (DSum (..))
+import PlutusPrelude
+import Untyped qualified as AgdaFFI
+import UntypedPlutusCore qualified as UPLC
+import UntypedPlutusCore.Transform.Simplifier
+
+-- | A class for types that can be unparsed to Agda code.
+class AgdaUnparse a where
+  agdaUnparse :: a -> String
+
+instance AgdaUnparse AgdaFFI.UTerm where
+  agdaUnparse =
+    \case
+      AgdaFFI.UVar n -> "(UVar " ++ agdaUnparse (fromInteger n :: Natural) ++ ")"
```

Comment:
Perhaps future work, but you can use pretty printing operators like `<+>` so that you don't need to worry about space.